src/main/java/org/opentripplanner/model/calendar/openinghours/OHCalendarBuilder.java

```diff
@@ -0,0 +1,320 @@
+        : untilDayOfWeek.getValue();
+      for (int i = fromDayOfWeek.getValue(); i <= untilAdjusted; i++) {
+        int dayValue = i > 7 ? i - 7 : i;
+        on(DayOfWeek.of(dayValue));
+      }
+      return this;
+    }
+
+    /**
+     * Sets every weekday in the range to be open on every instance they exist within the defined period
+     * on the defined month range. Both ranges are inclusive in both ends.
+     * If the builder is set be for times after midnight, the days are shifted one day forward
+     * so that first day of the first month is never on but the first day after the last month can be.
+     */
+    public OpeningHoursBuilder on(
+      Month fromMonth,
+      Month untilMonth,
+      DayOfWeek fromDayOfWeek,
+      DayOfWeek untilDayOfWeek
+    ) {
+      if (fromMonth == null || fromDayOfWeek == null) {
+        return this;
+      }
+
+      Set<Month> months = new HashSet<>();
+      if (untilMonth == null) {
+        months.add(fromMonth);
+      } else {
+        int untilMonthAdjusted = fromMonth.getValue() > untilMonth.getValue()
+          ? untilMonth.getValue() + 12
+          : untilMonth.getValue();
+        for (int i = fromMonth.getValue(); i <= untilMonthAdjusted; i++) {
+          int monthValue = i > 12 ? i - 12 : i;
+          months.add(Month.of(monthValue));
+        }
+      }
+
+      Set<DayOfWeek> daysOfWeek = new HashSet<>();
+      if (untilDayOfWeek == null) {
+        daysOfWeek.add(fromDayOfWeek);
+      } else {
+        int untilDayAdjusted = fromDayOfWeek.getValue() > untilDayOfWeek.getValue()
+          ? untilDayOfWeek.getValue() + 7
+          : untilDayOfWeek.getValue();
+        for (int i = fromDayOfWeek.getValue(); i <= untilDayAdjusted; i++) {
+          int dayValue = i > 7 ? i - 7 : i;
+          daysOfWeek.add(DayOfWeek.of(dayValue));
+        }
+      }
+
+      var dateToProcess = afterMidnight ? startOfPeriod.minusDays(1) : startOfPeriod;
+      int i = 0;
+      while (i < daysInPeriod) {
+        if (months.contains(dateToProcess.getMonth())) {
+          if (daysOfWeek.contains(dateToProcess.getDayOfWeek())) {
+            openingDays.set(i);
+          }
+          dateToProcess = dateToProcess.plusDays(1);
+          i += 1;
+        } else {
+          int daysToSkip =
+            YearMonth.of(dateToProcess.getYear(), dateToProcess.getMonth()).lengthOfMonth() -
+            dateToProcess.getDayOfMonth() +
+            1;
+          dateToProcess = dateToProcess.plusDays(daysToSkip);
+          i += daysToSkip;
+        }
+      }
+      return this;
+    }
+
+    /**
+     * Sets the days that are true in the given {@link BitSet} to be on without setting any days off.
+     * If the builder is set be for times after midnight, the days are not shifted in this case.
+     */
+    public OpeningHoursBuilder on(BitSet days) {
+      if (days.size() != openingDays.size()) {
+        return this;
+      }
+      openingDays.or(days);
+      return this;
+    }
```

Comment:
This method is used when parsing opening hours from OSM. The input is the days that are in common with some rules define that the park or whatever should be open and rules that define that it should not be open. These days don't necessarily follow some easily defined logic so therefore it's much easier to just do and operation on the BitSets of the two OpeningHoursBuilders and create a new OpeningHoursBuilder that is on the days that are in common. The input could be another OpeningHoursBuilder instead of the BitSet but I'm not sure if that is cleaner. I'm fully aware that some of this code is a bit "hacky" but it makes constructing the opening hours much easier based on the OSM rules. This and the other methods can't currently be private or package local because they are called from the OSM parser which is located in `org.opentripplanner.openstreetmap` package. However, it's possible that the parser could be moved to be in the same package as this (the parser is also used by one realtime park updater that uses OSM opening hours format).

Reply:
Decided to make these type of methods either private or to take in builder instead of the BitSet. I've implemented the changes now.